kernel/src/predicates/mod.rs

```diff
@@ -0,0 +1,453 @@
+use crate::expressions::{
+    BinaryOperator, Expression as Expr, Scalar, UnaryOperator, VariadicOperator,
+};
+use crate::schema::DataType;
+
+use std::cmp::Ordering;
+use tracing::debug;
+
+/// Evaluates a predicate expression tree against column names that resolve as scalars. Useful for
+/// testing/debugging but also serves as a reference implementation that documents the expression
+/// semantics that kernel relies on for data skipping.
+pub(crate) trait PredicateEvaluator {
+    type Output;
```

Comment:
I like the idea, but is there ever a chance we won't want every evaluation to return the same output?

Reply:
It already exists even before this PR: Delta data skipping maps `Expr` to `Option<Expr>` (and then applies the resulting skipping expression to engine data batches during log replay). Parquet footer skipping directly evaluates `Expr` to produce `Option<bool>`.